tests/functional/test_install.py

```diff
@@ -2030,6 +2030,24 @@ def test_install_yanked_file_and_print_warning(script, data):
     assert "Successfully installed simple-3.0\n" in result.stdout, str(result)
 
 
+def test_error_all_yanked_files_and_no_pin(script, data):
+    """
+    Test raising an error if there are only "yanked" files available and no pin
+    """
+    result = script.pip(
+        "install",
+        "simple",
+        "--index-url",
+        data.index_url("yanked_all"),
+        expect_error=True,
+    )
+    # Make sure an error is raised
+    assert (
+        result.returncode == 1
+        and "ERROR: No matching distribution found for simple\n" in result.stderr
+    ), str(result)
```

Comment:
Wouldn't this also pass without this patch? To make sure this change works as expected, we need a test that would fail if run without the change made in the factory.

Reply:
Just double checked, this will NOT pass without the patch: `FAILED tests/functional/test_install.py::test_error_all_yanked_files_and_no_pin - AssertionError: Script passed unexpectedly.`

Without the patch, pip would just raise a warning and install the latest yanked release, as reported in https://github.com/pypa/pip/issues/10617#issuecomment-953291018